src/mesa/drivers/dri/i965/intel_batchbuffer.c

```diff
@@ -742,11 +742,18 @@ submit_batch(struct brw_context *brw, int in_fence_fd, int *out_fence_fd)
    int ret = 0;
 
    if (batch->use_shadow_copy) {
-      void *bo_map = brw_bo_map(brw, batch->batch.bo, MAP_WRITE);
-      memcpy(bo_map, batch->batch.map, 4 * USED_BATCH(*batch));
+      void *state_map, *batch_map = brw_bo_map(brw, batch->batch.bo, MAP_WRITE);
+      if(batch_map == NULL)
+        return -1;
+
+      state_map = brw_bo_map(brw, batch->state.bo, MAP_WRITE);
+      if (state_map == NULL) {
+        brw_bo_unmap(batch->batch.bo);
+        return -1;
```

Comment:
Why would the mapping fail, what was the failing usecase/app?

Reply:
The problem happened only once and a core dump was the only thing the customer could provide, I found the the application tried to access a NULL pointer and triggered A/V. To answer your question: we don't know the root cause of the mapping failure,

This is simply a w/a to fail the submission in case of mapping failure, hopefully  the user app checks the error and fails gracefully instead of crashes to terminal.